src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -4866,5 +4866,5 @@ TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, SingleDeinitializeCallOnRelo
     ASSERT_EQ(factory.reloadDefinition("my_new_pipeline", std::move(info), std::move(connections), manager), StatusCode::OK);
     // Each custom node has effectively 1 internalManager initialized, because they use same library instance
     // in order to count whether deinitialize has been called expected number of times
-    ASSERT_EQ(LibraryCountDeinitialize::deinitializeCounter, 1);
+    ASSERT_EQ(LibraryCountDeinitialize::deinitializeCounter, 3);
```

Comment:
Why is it 3?

Reply:
It is 3 because as if for now, because calling reload on pipelinedefinition causes all nodeResources to be replaced/erased. Therefore, when we effectively remove 1 of 3 custom_nodes in pipeline deinitialize is called 3 times.
One deinitialize call is caused by erasing the resource allocated for the removed node.
Two deinitialize calls are caused by creating new customNodeLibraryInternalManager instances, which replace the old ones. Since there are none executions active replacing shared_ptr causes CNLIMWrapper to call its destructor.

Reply:
Mind that this test is going to be changed in `cleanup mechanism` task.